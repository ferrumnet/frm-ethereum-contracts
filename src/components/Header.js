/* eslint-disable jsx-a11y/anchor-is-valid */
import React from "react";
import { NavLink, withRouter } from "react-router-dom";

const Header = ({
  history: {
    location: { pathname }
  }
}) => {
  const isAdmin = pathname.match(/admin/g) ? true : false;

  const activeStyle = { color: "#FFFFFF" };
  return (
    <header>
      <nav className="navbar  navbar-expand-lg navbar-dark py-4 bg-dark">
        <NavLink className="navbar-brand" to="/">
          Ferrum Contracts
        </NavLink>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarText"
          aria-controls="navbarText"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarText">
          <ul className="navbar-nav mr-auto ">
            {isAdmin && (
                <li className="nav-item">
                  <NavLink
                    className="nav-link"
                    exact
                    to="/admin/deploy"
                    activeStyle={activeStyle}
                  >
                    Deploy
                  </NavLink>
                </li>

            )}
          </ul>
          <ul className="navbar-nav ml-auto">
            <li className="nav-item dropdown">
              <a
                className="nav-link dropdown-toggle"
                id="navbarDropdownMenuLink-4"
                data-toggle="dropdown"
                aria-haspopup="true"
                aria-expanded="false"
              >
                <i className="fas fa-user"></i> Account
              </a>
              <div
                className="dropdown-menu dropdown-menu-right p-2"
                aria-labelledby="navbarDropdownMenuLink-4"
              >
                <NavLink
                  className="dropdown-item"
                  exact
                  to="/admin/deploy"
                  activeStyle={activeStyle}
                >
                  Admin
                </NavLink>
                <NavLink
                  className="dropdown-item"
                  exact
                  to="/"
                  activeStyle={activeStyle}
                >
                  Customer
                </NavLink>
              </div>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  );
};

export default withRouter(Header);
