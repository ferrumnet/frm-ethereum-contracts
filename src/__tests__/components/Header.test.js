import Header from "../../components/Header";
import { shallow } from "enzyme";
import React from "react";

describe("Test Header Component", () => {
  test("It should match the snapshot", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
