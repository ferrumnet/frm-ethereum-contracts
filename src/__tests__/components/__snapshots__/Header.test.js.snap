// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header Component It should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Header) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "_threadCount": 0,
        "displayName": "Router",
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_threadCount": 0,
          "displayName": "Router",
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
